recipes/dolphin/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "dolphin" %}
+{% set version = "0.1.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/opera-adt/dolphin/archive/v{{ version }}.tar.gz
+  sha256: fe71afb65d56e285a129bd55fe5b113c34cb04833db36b8337e2c59247e0351d
+
+build:
+  entry_points:
+    - dolphin = dolphin.cli:main
+  noarch: python
+  script:
+    - export SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
+    - {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - setuptools >=64.0
+    - setuptools-scm >=6.2
+    - pip >=21.3
+    - git  # for pip install, due to setuptools_scm
+  run:
+    - python >=3.8
+    - gdal >=3.3
+    - h5py >=3.6
+    - hdf5 <1.12.2 # https://github.com/SciTools/iris/issues/5187 and https://github.com/pydata/xarray/issues/7549
```

Comment:
That issue seems to be in the lnetcdf-c and not hdf5. Also, we are shipping a patched version of netcdf-c with the upstream fix so you probably don't need that pin.